it/uk/gov/hmrc/transitmovementsrouter/connectors/EISConnectorSpec.scala

```diff
@@ -129,43 +146,104 @@ class EISConnectorSpec
         // If a 500 error is returned, this most likely means a retry happened, the first place to look
         // should be the code the determines if a result is successful.
 
+        val expectedConversationId = ConversationId(movementId, messageId)
+
         def stub(currentState: String, targetState: String, codeToReturn: Int) =
           server.stubFor(
             post(
               urlEqualTo(uriStub)
             )
               .inScenario("Standard Call")
               .whenScenarioStateIs(currentState)
+              .withHeader("Date", anything)
               .withHeader("X-Correlation-Id", matching(RegexPatterns.UUID))
+              .withHeader("X-Conversation-Id", equalTo(expectedConversationId.value.toString))
               .withHeader("CustomProcessHost", equalTo("Digital"))
               .withHeader(HeaderNames.ACCEPT, equalTo("application/xml"))
+              .withHeader(HeaderNames.CONTENT_TYPE, equalTo("application/xml"))
               .willReturn(aResponse().withStatus(codeToReturn))
```

Comment:
Not sure. But do we need to test with some dummy Authorisation header?

Reply:
This is a bit of a weird one to think about, but the answer is no.

What we're doing in the tests in this service is testing how our system reacts to certain scenarios. In these integration tests, we are saying "we expect these headers, with these values". What we're really doing that is testing that the receiving function is getting what we asked for -- in this case, we have tested that the token we have set in the config has ended up hitting the dummy endpoint.

Now, we define how the endpoint behaves, so in the lines you highlighted, we say "if we get these headers, then respond in this way". We can respond how we like to test the behaviour to the response. 

So, we don't need to test with a dummy header, because that adds nothing -- we know that whatever we put into the config is output here. We need to test how different responses are handled, and do it in the simplest way -- in this case, same input, different response.